packages/@react-aria/autocomplete/src/useAutocomplete.ts

```diff
@@ -350,7 +350,7 @@ export function useAutocomplete(props: AriaAutocompleteOptions, state: Autocompl
       autoCorrect: 'off',
       // This disable's the macOS Safari spell check auto corrections.
       spellCheck: 'false',
-      [parseInt(React.version, 10) >= 17 ? 'enterKeyHint' : 'enterkeyhint']: 'enter',
+      enterKeyHint: 'go',
```

Comment:
@devongovett Do we need to apply the same kind of logic change to useDateSegment?

Reply:
no, the change here was because it is being passed into `useTextField` which only accepts `enterKeyHint` not `enterkeyhint`. `useDateSegment` props are being passed directly to a dom element.